index.html

```diff
@@ -416,6 +424,13 @@
       the <a>context object</a>'s <a>next sibling</a>.
     </dl>
   </ol>
+  
+  <p class=note>No special handling for <code><a>template</a></code> elements is included in the 
+  above "<code>afterbegin</code>" and "<code>beforeend</code>" cases. Unlike 
+  <a for="Element">innerHTML</a>, <a for="Element">insertAdjacentHTML</a> will insert the given 
+  <var>fragment</var> into a <code><a>template</a></code>'s <a>children</a> rather than into its
+  <a>template content</a>.</p>
```

Comment:
While I also find this a bit odd, I checked Chrome, Edge, and Firefox (if you could check Safari, that'd be great!). None of them do any special case handling here. At this point, I'm content to document the existing interoperable behavior.